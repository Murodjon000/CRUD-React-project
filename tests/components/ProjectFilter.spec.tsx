import shallowSetup from "./mock.spec";
import toJson from "enzyme-to-json";

test("should render ProjectFilter component", () => {
  const { projectFilterWrapper } = shallowSetup();

  expect(
    projectFilterWrapper.find("div").first().hasClass("filter__wrapper")
  ).toBe(true);
  expect(projectFilterWrapper.find("select").hasClass("test-filter")).toBe(
    true
  );
  expect(projectFilterWrapper.find("option").first().text()).toBe("All");
});

test("should render ProjectFilter component Snapshot", () => {
  const { projectFilterWrapper } = shallowSetup();

  expect(toJson(projectFilterWrapper)).toMatchSnapshot();
});
