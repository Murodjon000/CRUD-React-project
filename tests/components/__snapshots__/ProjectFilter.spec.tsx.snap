// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Details component Snapshot 1`] = `
<div
  className="project__details-wrapper"
>
  <div
    className="project__details-wrapper-image"
  >
    <img
      alt=""
      className="test-img"
      src="https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg"
    />
  </div>
  <div
    className="project__details-wrapper-box"
  >
    <h1
      className="test-title"
    >
      Where can I get some?
    </h1>
    <p
      className="test-desc"
    >
       
      ðŸ‘€ PageMaker including versions of Lorem Ipsum.
    </p>
    <button
      className="test-btn"
      onClick={[Function]}
      type="button"
    >
      Back to home
    </button>
  </div>
</div>
`;

exports[`should render ProjectFilter component Snapshot 1`] = `
<div
  className="filter__wrapper"
>
  <select
    className="test-filter"
    name="category"
    onChange={[Function]}
    value="All"
  >
    <option
      key="All"
      value="All"
    >
      All
    </option>
    <option
      key="High"
      value="High"
    >
      High
    </option>
    <option
      key="Medium"
      value="Medium"
    >
      Medium
    </option>
    <option
      key="Low"
      value="Low"
    >
      Low
    </option>
  </select>
</div>
`;
